src/core/engines/full-margin-v2/FullMarginMathV2.sol

```diff
@@ -141,28 +141,29 @@ library FullMarginMathV2 {
 
         (strikes, weights, syntheticUnderlyingWeight, intrinsicValue) = convertPutsToCalls(_detail);
 
-        pois = createPois(_detail.putStrikes, strikes, _detail.spotPrice);
+        pois = createPois(strikes, _detail.putStrikes.length, _detail.spotPrice);
 
         payouts = calcPayouts(
             PayoutsParams(pois, strikes, weights, syntheticUnderlyingWeight, _detail.spotPrice, intrinsicValue)
         );
     }
 
     function createPois(
-        uint256[] memory putStrikes,
         uint256[] memory strikes,
+        uint256 numOfPuts,
         uint256 spotPrice
     ) private pure returns (uint256[] memory pois) {
         uint256 epsilon = spotPrice / 10;
 
-        bool hasPuts = putStrikes.length > 0;
+        bool hasPuts = numOfPuts > 0;
 
         // left of left-most + strikes + right of right-most
         uint256 poiCount = (hasPuts ? 1 : 0) + strikes.length + 1;
 
         pois = new uint256[](poiCount);
 
-        if (putStrikes.length > 0) pois[0] = strikes.min() - epsilon;
+        // this line will revert if strike is lower than 10% of spot
+        if (hasPuts) pois[0] = strikes.min() - epsilon;
```

Comment:
This is getting addressed in leo's PR. We are now setting epsilon to `strikes.min() / 10`